os/kernel/binary_manager/binary_manager_binfile.c

```diff
@@ -116,8 +118,13 @@ void binary_manager_scan_ubin(void)
 				if (ret < 0) {
 					continue;
 				}
-				/* If binary is not registered, register it */
-				(void)binary_manager_register_ubin(header_data.bin_name);
+				bin_idx = binary_manager_get_index_with_name(header_data.bin_name);
+				if (bin_idx < 0) {
+					/* If binary is not registered, register it */
+					(void)binary_manager_register_ubin(header_data.bin_name, header_data.bin_priority);
+				} else if (header_data.bin_priority > BM_PRIORITY_MAX) {
+					BIN_PRIORITY(bin_idx) = header_data.bin_priority;
```

Comment:
1. Is there any chance to be reach here even this function is called at start of binary manager once?
2. Is it enough to save priority only? When do you write other information?

Reply:
For 2nd, we could reduce the duplicated operation.
solution1.
1. scan binary at start of binary manager and at binary update and update the binary management table
2. load binary using above

solution2.
1. no update binary management table
2. update all of information when load binary

reason
1. Binary header which includes binary information is in flash so that it will not be changed
  => no problem with updating the table at early time
2. no need to read binary header twice for performance